deepchem/utils/fake_data_generator.py

```diff
@@ -0,0 +1,157 @@
+"""
+A fake data generator
+"""
+import random
+import numpy as np
+from deepchem.data import NumpyDataset
+from deepchem.feat import GraphData
+
+
+class FakeGraphGenerator:
+  """Generates a random graphs which can be used for testing or other purposes.
+
+  Example
+  -------
+  >>> from deepchem.utils.fake_data_generator import FakeGraphGenerator
+  >>> fgg  = FakeGraphGenerator(avg_n_nodes=8, n_node_features=5, avg_degree=4, n_edge_features=3, n_classes=2, task='graph', z=5)
+  >>> graphs = fgg.sample(n_graphs=10)
+  >>> type(graphs)
+  <class 'deepchem.data.datasets.NumpyDataset'>
+  >>> type(graphs.X[0])
+  <class 'deepchem.feat.graph_data.GraphData'>
+  >>> len(graphs) == 10  # num_graphs
+  True
+
+  Note
+  ----
+  This fake data generator is based on torch_geometric.dataset.FakeDataset
+  class. The reason for having DeepChem's own implementation are two fold:
+  1) The current stable version (2.0.3) of torch_geometric's FakeDataset
+  doesn't support global features and kwargs.
+  2) The generated dataset in FakeDataset have same node feature distribution
+  irrespective of graph or node label which make it unsuitable to use during testing.
+  """
+
+  # TODO @arunppsg Once torch_geometric's version 2.0.4 is release which
+  # fixes above two issues, this can be depreciated.
+  def __init__(self,
+               avg_n_nodes: int = 10,
+               n_node_features: int = 5,
+               avg_degree: int = 4,
+               n_edge_features: int = 3,
+               n_classes: int = 2,
+               task: str = 'graph',
+               **kwargs):
+    """
+    Parameters
+    ----------
+    avg_n_nodes: int, default 10
+      Average number of nodes in a graph
+    n_node_features: int, default 5
+      Average number of node features in a graph
+    avg_degree: int, default 4
+      Average degree of the graph
+    n_edge_features: int, default 3
+      Average number of features in the edge
+    task: str, default 'graph'
+      Indicates node-level labels or graph-level labels
+    kwargs: optional
+      Additional graph attributes and their shapes , e.g. `global_features = 5`
+    """
+    self.avg_n_nodes = max(avg_n_nodes, avg_degree)
+    self.avg_degree = max(avg_degree, 1)
+    self.n_node_features = n_node_features
+    self.n_edge_features = n_edge_features
+    self.n_classes = n_classes
+    self.task = task
+    self.kwargs = kwargs
+
+  def sample(self, n_graphs: int = 100) -> NumpyDataset:
+    """Samples graphs
+
+    Parameters
+    ----------
+    n_graphs: int, default 100
+      Number of graphs to generate
+
+    Returns
+    -------
+    graphs: NumpyDataset
+      Generated Graphs
+    """
+    graphs, labels = [], []
+    for i in range(n_graphs):
+      if self.task == 'graph':
+        label = random.randint(0, self.n_classes - 1)
+      elif self.task == 'node':
+        # TODO
```

Comment:
Can you add an explanation of the TODO here?

Reply:
I implemented the TODO part.